File: app/routers/insight.py
from typing import List, Optional

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import get_db
from app.crud import insight as crud_insight
from app.schemas.insight import (
    TotalCountResponse,
    TopPerformerResponse,
    DailyTrendResponse,
    CategoryDistributionResponse,
    CompareInfluencerResponse,
)


router = APIRouter(prefix="/insights", tags=["Insights"])


@router.get("/total-selections", response_model=TotalCountResponse)
def get_total_selections(db: Session = Depends(get_db)):
    total = crud_insight.get_total_selections(db)
    return {"total": total}


@router.get("/top-performer", response_model=Optional[TopPerformerResponse])
def get_top_performer(db: Session = Depends(get_db)):
    return crud_insight.get_top_performer(db)


@router.get("/total-influencers", response_model=TotalCountResponse)
def get_total_influencers(db: Session = Depends(get_db)):
    total = crud_insight.get_total_influencers(db)
    return {"total": total}


@router.get("/daily-trends", response_model=List[DailyTrendResponse])
def get_daily_trends(db: Session = Depends(get_db)):
    return crud_insight.get_daily_trends(db)


@router.get("/category-distribution", response_model=List[CategoryDistributionResponse])
def get_category_distribution(db: Session = Depends(get_db)):
    return crud_insight.get_category_distribution(db)


@router.get("/compare", response_model=List[CompareInfluencerResponse])
def compare_influencers(i1: int, i2: int, db: Session = Depends(get_db)):
    return crud_insight.compare_influencers(db, i1, i2)

File: frontend/src/api/insight.ts
const BASE_URL = "http://localhost:8000/insights";

// 1. 전체 선택 수
export async function getTotalSelections() {
  const res = await fetch(`${BASE_URL}/total-selections`);
  if (!res.ok) throw new Error("total-selections 실패");
  return res.json();
}

// 2. 인기 인플루언서
export async function getTopPerformer() {
  const res = await fetch(`${BASE_URL}/top-performer`);
  if (!res.ok) throw new Error("top-performer 실패");
  return res.json();
}

// 3. 전체 인플루언서 수
export async function getTotalInfluencers() {
  const res = await fetch(`${BASE_URL}/total-influencers`);
  if (!res.ok) throw new Error("total-influencers 실패");
  return res.json();
}

// 4. 일별 트렌드
export async function getDailyTrends() {
  const res = await fetch(`${BASE_URL}/daily-trends`);
  if (!res.ok) throw new Error("daily-trends 실패");
  return res.json();
}

// 5. 카테고리 분포
export async function getCategoryDistribution() {
  const res = await fetch(`${BASE_URL}/category-distribution`);
  if (!res.ok) throw new Error("category-distribution 실패");
  return res.json();
}

// 6. 인플루언서 비교
export async function compareInfluencers(i1: number, i2: number) {
  const res = await fetch(
    `${BASE_URL}/compare?i1=${i1}&i2=${i2}`
  );
  if (!res.ok) throw new Error("compare 실패");
  return res.json();
}
